monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/HttpClientPoolBuilder.java

```diff
@@ -52,15 +53,11 @@ public HttpClientPoolBuilder withDefaultScheduledExecutorService() {
         return this;
     }
 
-    public HttpClientPoolBuilder withSingleHostHttpClientBuilder(final SingleHostHttpClientBuilder singleHostHttpClientBuilder) {
+    public HttpClientPoolBuilder withSingleHostHttpClientBuilder(final Function<InetAddress, SingleHostHttpClientBuilder> singleHostHttpClientBuilder) {
```

Comment:
What's the reason for making it a function? Why not pass `InetAddress` as a building parameter separately?

Reply:
it is a mandatory parameter coming from the current builder so I'm not sure to understand what's needed ?

Reply:
I propose to introduce a new object variable `InetAddress inetAddress` and having two arguments in the builder method

Reply:
It needs to be dynamically set in the HTTP pool for failover of DNS.